libbpfgo.go

```diff
@@ -760,19 +760,16 @@ func (b *BPFMap) GetValueBatch(keys unsafe.Pointer, startKey, nextKey unsafe.Poi
 		Flags:     C.BPF_ANY,
 	}
 
-	errC := C.bpf_map_lookup_batch(b.fd, startKey, nextKey, keys, valuesPtr, &countC, bpfMapBatchOptsToC(opts))
-	if errC != 0 {
-		sc := syscall.Errno(-errC)
-		if sc != syscall.EFAULT {
-			if uint32(countC) != count {
-				return collectBatchValues(values, uint32(countC), b.ValueSize()),
-					fmt.Errorf("failed to retrieve ALL elements in map %s, fetched (%d/%d): %w", b.name, uint32(countC), count, sc)
-			}
-		}
-		return nil, fmt.Errorf("failed to batch lookup values %v in map %s: %w", keys, b.name, syscall.Errno(-errC))
+	ret, errC := C.bpf_map_lookup_batch(b.fd, startKey, nextKey, keys, valuesPtr, &countC, bpfMapBatchOptsToC(opts))
+	processed := uint32(countC)
+
+	if ret != 0 && errC != syscall.ENOENT {
+		// ret = -1 && errno == syscall.ENOENT indicates a partial read.
```

Comment:
In this case, is it a partial failure or complete? Can you clarify the comment a little more?

Reply:
In this condition, there's a "partial success", we have processed some of the elements, but not as many as requested.

Happy to add more context here. Did not want to duplicate the big comment in `GetValueAndDeleteBatch`, that's why I referred to it in this method's doc

Reply:
We do that below with `return collectBatchValues(values, processed, b.ValueSize())`. Either with a partial or a complete success we return there.

What I meant before is that `ret = -1 && errno == syscall.ENOENT` means partial success and we won't error in this case, we will return the subset of the elements that got read. This usually means that the requested number of elements was higher than the existing number of elements. 

Otherwise, we return with an error in line 768, and this means that there is a legit error and nothing got processed